lec01-introduction.Rmd

```diff
@@ -2,3 +2,176 @@
+
+And if you think it sounds complicated to remember all of these functions by heart, don't worry there are several places to get help. The one I use the most is called tab-complete: often I remember the beginning of the function I want to use, but I don't remember exactly what it was called. For example, if I want to compute the median of a set of numbers, and I don't remember if the function for this was just `med` or `median`, I can just type the first three letters, hit tab <!-- Technically twice by default in the console, but since it is once in Rstudio, lets skip that distinction -->  and all functions starting with those letters. Here I can see that it is probably `median` I want to use, and to be sure I can type `?median` and read what it does. I will show you a way to read these help pages with some nicer formatting later in this lecture.
+
+In my experience, learning programming really is similar to learning a foreign language - you will often learn the most from just trying to do something and receiving feedback (from the computer or another person)! When there is something you can't wrap you head around, or if you are actively trying to find a new way of expressing a thought, then look it up, just as you would with a natural language.
+
+## Programming in R
+
+### The basics
+With that background to some of the concepts of programming, let's compare the workflow of analyzing a data set in R vs in a Excel. _Pull up Excel and R console next to each other._ In Excel, we open a file via 'File -> Open -> Navigate to file'. You can conceptualize this as if there is a function `file.open()` for which you need to specify a file location, (`file.open(/home/joel/iris.csv`). In R this looks similar, it is `read.csv('/home/joel/iris.csv')`, <!--need to write in windows paths...--> so it is pretty much the same, slightly different words are used and you type them in instead of clicking your way to them.
+
+When you open a file in Excel, it will immediately display the content in the window. Likewise, the R console will display the information of the data set when you read it in. This sample data set that we are using for this class describes the length and width of sepals and petals for three species of iris flowers.
+
+```{r}
+read_csv('/home/joel/iris.csv') # need to dry run this with windows path names...
+```
+
+
+If we want to calculate something in Excel, we would highlight the column and then click mean... <!-- I actually don't have Excel and don't know how to do this, will check tomorrow... --> In R we first give the data a nick name where it is stored, so instead of printing out to console, these values are stored in this R session and we can access it by this name `iris`, (technically we say that we have assigned it to variable `iris`).
+
+```{r}
```

Comment:
Same as above, need to use `eval=FALSE`.

If you do want to include it, put the file in the repo and change to `read_csv('iris.csv')`

Reply:
But I need it for all my plots... and we are including the plots in the lessons files right? i will just upload the csv to the repo and change the path

Reply:
No, don't include the plots. They will be automatically generated from the Rmd file via Travis.

Reply:
Yes, sorry that's what I meant. If I would set `eval=FALSE`, then the `iris` variable would not have been created and Travis could not create my plots, right?